tests/unittests/core/worker/test_producer.py

```diff
@@ -139,10 +140,12 @@ def test_update_and_produce(producer, database, random_dt):
     assert producer.naive_algorithm.algorithm._suggested == possible_values
 
 
-def test_register_new_trials(producer, database, random_dt):
+def test_register_new_trials(producer, random_dt):
```

Comment:
Why not storage fixture instead of get_storage()?

Reply:
Yeah... there's a weird bug about this. If I use the `storage` fixture, the test fails, because the storage is empty, a bit as if the fixture was run a second time after `hacked_exp`, but I tried to check and I don't think it is? Either way I suspect fixture shenanigans and I didn't feel like debugging that, but I could look into it if you want, after everything else is fixed.

Reply:
ok, let's see if time permits, otherwise we keep it that way.